security-analysis-ws/src/main/java/eu/itesla_project/security/rest/api/impl/SecurityAnalysisServiceImpl.java

```diff
@@ -0,0 +1,205 @@
+/**
+ * Copyright (c) 2016, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package eu.itesla_project.security.rest.api.impl;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.StreamingOutput;
+
+import org.jboss.resteasy.plugins.providers.multipart.InputPart;
+import org.jboss.resteasy.plugins.providers.multipart.MultipartFormDataInput;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import eu.itesla_project.commons.io.table.CsvTableFormatterFactory;
+import eu.itesla_project.security.LimitViolationFilter;
+import eu.itesla_project.security.LimitViolationType;
+import eu.itesla_project.security.Security;
+import eu.itesla_project.security.SecurityAnalysisResult;
+import eu.itesla_project.security.SecurityAnalyzer;
+import eu.itesla_project.security.json.SecurityAnalysisResultSerializer;
+import eu.itesla_project.security.rest.api.SecurityAnalysisService;
+
+/**
+ *
+ * @author Giovanni Ferrari <giovanni.ferrari@techrain.it>
+ */
+public class SecurityAnalysisServiceImpl implements SecurityAnalysisService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SecurityAnalysisServiceImpl.class);
+
+    private enum Format {
```

Comment:
I leaved the enum separated because the web service and the itools command  could support different output formats in the future..